modules/tinymce/src/core/main/ts/mode/Readonly.ts

```diff
@@ -156,14 +173,20 @@ const processReadonlyEvents = (editor: Editor, e: Event): void => {
 };
 
 const registerReadOnlySelectionBlockers = (editor: Editor): void => {
-  editor.on('ShowCaret', (e) => {
+  editor.on('beforeinput paste', (e) => {
     if (isReadOnly(editor)) {
       e.preventDefault();
     }
   });
 
-  editor.on('ObjectSelected', (e) => {
-    if (isReadOnly(editor)) {
+  editor.on('BeforeSetContent', (e) => {
+    if (isReadOnly(editor) && editor.mode.isSelectionEnabled() && !e.initial) {
```

Comment:
Just trying to determine if `editor.mode.isSelectionEnabled()` is required in the check here? What is this new code trying to achieve?

Reply:
We switched from `contentEditable=true` when the editor is in this `selectionEnabled` mode, so you can place a cursor and move around the editor with only keyboard. What I'm trying to achieve is blocking `setContent/insertContent` when the cursor is placed in the editor, so that only annotating is permitted but not altering the content of the editor. The previous readonly (or full readonly mode) disables the editor body `contentEditable='false'` to the body, so inserting content is blocked by default.

Reply:
Ah right. I think I know what you are trying to do now. Could add a comment clarifying the purpose of the check maybe?